moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -401,8 +408,11 @@ void CurrentStateMonitor::tfCallback()
         continue;
       }
 
+      if (auto joint_time_it = joint_time_.find(joint); joint_time_it == joint_time_.end())
+        joint_time_.emplace(joint, rclcpp::Time(0, 0, RCL_ROS_TIME));
+
       // allow update if time is more recent or if it is a static transform (time = 0)
-      if (latest_common_time <= joint_time_[joint] && latest_common_time > rclcpp::Time(0))
+      if (latest_common_time <= joint_time_[joint] && latest_common_time > rclcpp::Time(0, 0, RCL_ROS_TIME))
         continue;
       joint_time_[joint] = latest_common_time;
```

Comment:
`// allow update if time is more recent or if it is a static transform (time = 0)`

This block confuses me. 
- `latest_common_time` is always going to be 0, so seems like a misnomer
- because `latest_common_time` doesn't change, the second condition in the line 415 if statement will always be false, making the if never trigger. So why do we have it?

Reply:
How it's not changing .? if you look after few lines we set it to `transf.header.stamp`

Reply:
Bizarre, I thought I had everything expanded and ctrl + f still wasn't yielding anything.